commands/rm/main.cpp

```diff
@@ -0,0 +1,161 @@
+#include "command.hpp"
+
+#include <irods/rodsClient.h>
+#include <irods/connection_pool.hpp>
+#include <irods/filesystem.hpp>
+#include <irods/irods_exception.hpp>
+
+#include <boost/program_options.hpp>
+#include <boost/dll.hpp>
+#include <boost/progress.hpp>
+
+#include "experimental_plugin_framework.hpp"
+
+#include <iostream>
+#include <string>
+
+namespace fs = irods::experimental::filesystem;
+namespace po = boost::program_options;
+namespace ia = irods::experimental::api;
+
+namespace {
+    std::atomic_bool exit_flag{};
+
+    void handle_signal(int sig)
+    {
+        exit_flag = true;
+    }
+}
+
+
+
+namespace irods::cli
+{
+    void print_progress(const std::string& p)
+    {
+        static boost::progress_display prog{100};
+        try {
+            auto x = std::stoi(p.c_str(), nullptr, 10);
+            while(prog.count() != x) {
+                ++prog;
+            }
+        }
+        catch(...) {
+        }
+    } // print_progress
+
+    class rm : public command
+    {
+    public:
+        auto name() const noexcept -> std::string_view override
+        {
+            return "rm";
+        }
+
+        auto description() const noexcept -> std::string_view override
+        {
+            return "Removes collections or data objects";
+        }
+
+        auto help_text() const noexcept -> std::string_view override
+        {
+            auto help =R"(
+Given a fully qualfied or relative path, remove the collection or object
+
+irods rm [options] fully_qualified_logical_path
+
+      --unregister        : unregister data instead of unlinking data
+      --no_trash          : do no move items to the trash can
+      --number_of_threads : number of threads to use in recursive operations
+      --progress          : request progress as a percentage)";
+            return help;
+
+        }
+
+        auto execute(const std::vector<std::string>& args) -> int override
+        {
+            signal(SIGINT,  handle_signal);
+            signal(SIGHUP,  handle_signal);
+            signal(SIGTERM, handle_signal);
+
+            bool progress_flag{false}, no_trash{false}, unregister{false};
+            int thread_count{4};
+
+            using rep_type = fs::object_time_type::duration::rep;
+
+            po::options_description desc{""};
+            desc.add_options()
+                ("logical_path", po::value<std::string>(), "logical path to collection or object to remove")
+                ("unregister", po::bool_switch(&unregister), "unregister data instead of unlinking data")
+                ("no_trash", po::bool_switch(&no_trash), "do no move items to the trash can")
```

Comment:
trash can - perhaps we just call the it the trash?

Reply:
this reflects what is in file system, we can take a uniform pass over the whole thing if necessary.